test/request.test.ts

```diff
@@ -691,4 +691,61 @@ x//0u+zd/R/QRUzLOw4N72/Hu+UG6MNt5iDZFCtapRaKt6OvSBwy8w==
       expect(response.data).toEqual("ok");
     });
   });
+
+  it("@octokit/rest.js/issues/1497/error-messages-on-validation-error", function() {
+    const mock = fetchMock.sandbox().mock(
+      "https://request-errors-test.com/repos/gr2m/sandbox/branches/gr2m-patch-1/protection",
+      {
+        status: 403,
```

Comment:
The code is `400`, not `403`

Reply:
Sir, there are some errors like
```js
{
"resource":"Search",
    "field":"q",
    "code":"invalid"
}
```

How to handle this cases?

Reply:
Sir, why is code different here. For the previous test you told it should be `403`.

Reply:
> Sir, there are some errors like
> 
> ```js
> {
> "resource":"Search",
>     "field":"q",
>     "code":"invalid"
> }
> ```
> 
> How to handle this cases?

Sir for now I am handling these cases as:

"{ resource: Search, field: q, code: invalid }"

Reply:
Unfortunately the server responses are not consistent here. Instead of the custom `mapErrorToString` method, I would just use `JSON.stringify()`, to keep the bundle size low.

Reply:
Okk sir. I would make that change. Please tell me if any further changes are required?

Reply:
It looks good otherwise

Reply:
> It looks good otherwise

Thanks Sir.